liquidity/threshold_rule.go

```diff
@@ -63,6 +63,51 @@ func (r *ThresholdRule) validate() error {
 	return nil
 }
 
+// suggestSwap suggests a swap based on the liquidity thresholds configured.
+func (r *ThresholdRule) suggestSwap(channel *balances,
+	outRestrictions *Restrictions) (*LoopOutRecommendation, error) {
+
+	// Examine our total balance and required ratios to decide whether we
+	// need to swap.
+	amount, swapType := shouldSwap(
+		channel, r.MinimumInbound, r.MinimumOutbound,
+	)
+
+	// If a nil swap type is returned, we do not need to recommend any swaps
+	// so we return an empty swap set.
+	if swapType == nil {
+		return nil, nil
+	}
+
+	// We currently only support loop out, so return nil if a loop in is
+	// recommended.
+	switch *swapType {
+	case swap.TypeOut:
```

Comment:
Technically the code could be simplified further to return just a bool from shouldswap instead of a swap type.